### PR TITLE
Robot WiFi

### LINKED ISSUE #14: Setup WiFi hotspot
Astwifid and hosted are already installed, but need to be configured.

- Network Bridge
- `ip_forwarding=1` sysctl
- nftables configuration
- dnsmasq installation and configuration

### LINKED ISSUE #15: SSH Access
We should enable SSH access over WiFi. Dropbear is already installed.

Requires #14

### LINKED ISSUE #17: Install HTTP Server
We need an HTTP server for the WiFi interface. I would suggest `lighttpd`

### PR DESCRIPTION
**Pull request by raccube**
Closes #14 
Closes #15
Closes #17

### CODE REVIEW

File: recipes-robot/srobo-firewall/srobo-firewall.bb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+SUMMARY = "Package for SR nftables configuration"
+inherit systemd
+
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=691d84ab639c4b8173db302930bf71f4"
+
+SYSTEMD_AUTO_ENABLE = "enable"
+SYTEMD_SERVICE:${PN} = "nftables.service"
```
Comment (trickeydan) [inline recipes-robot/srobo-firewall/srobo-firewall.bb]:
```suggestion
SYSTEMD_SERVICE:${PN} = "nftables.service"
```

File: recipes-robot/srobo-kit/srobo-kit.bb [hunk 2]
```diff
@@ -26,4 +30,12 @@ do_install () {
...
     install -d ${D}/etc/modprobe.d/
     install -m 0644 ${WORKDIR}/exynos-blacklist.conf ${D}/etc/modprobe.d/
+    install -d ${D}/etc/dnsmasq.d/
+    install -m 0644 ${WORKDIR}/dnsmasq.conf ${D}/etc/dnsmasq.d/srobo.conf
+    install -d ${D}/etc/systemd/network
+    install -m 0644 ${WORKDIR}/br0.netdev ${D}/etc/systemd/network/br0.netdev
+    install -m 0644 ${WORKDIR}/br0.network ${D}/etc/systemd/network/br0.network
+    install -m 0644 ${WORKDIR}/eth0.network ${D}/etc/systemd/network/eth0.network
+    install -d ${D}/etc/sysctl.d
+    echo "net.ipv4.ip_forward=1" > ${D}/etc/sysctl.d/ip_forward.conf
```
Comment (trickeydan) [inline recipes-robot/srobo-kit/srobo-kit.bb]: Nit: Might be worth moving this config to a separate `srobo-network` recipe just to stop the srobo-kit recipe getting too complex.

**Review by trickeydan — CHANGES_REQUESTED** [review]
Looks good. Requesting changes on typo.

Please don't merge until @Tyler-Ward is happy.

File: recipes-robot/srobo-kit/srobo-kit.bb (hunk 2 above)
Comment (Tyler-Ward) [reply recipes-robot/srobo-kit/srobo-kit.bb]: Putting the network config in a separate recipe would seem sensible. given how much of the kit recipe is now networking

File: recipes-robot/srobo-firewall/srobo-firewall.bb (hunk 1 above)
Comment (Tyler-Ward) [reply recipes-robot/srobo-firewall/srobo-firewall.bb]: I assume given this typo that this has not yet been tested?

**Comment by Tyler-Ward** [comment]
I can confirm that this builds on the build pi however I haven't tested it on a kit

**Review by trickeydan — APPROVED** [review]

**Review by Tyler-Ward — APPROVED** [review]
The new version also builds so assuming this works on a kit lets get this merged

**Pull request merged**